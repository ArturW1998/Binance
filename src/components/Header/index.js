import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../../imgs/logo.svg';

const Header = () => (
  <header>
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <Link className="navbar-brand" to="/">
        <img src={logo} height="25" alt="Binance Logo" />
      </Link>

      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>

      <div className="collapse navbar-collapse" id="navbarNav">
        <span className="small text-white">Trade history viewer web app</span>
      </div>
    </nav>
  </header>
);

export default Header;
