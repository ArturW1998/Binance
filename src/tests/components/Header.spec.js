import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

describe('Header component', () => {
  const header = shallow(<Header />);

  it('renders properly', () => {
    expect(header.debug()).toMatchSnapshot();
  });

  it('renders header', () => {
    expect(header.find('header')).toHaveLength(1);
  });

  it('renders navbar text', () => {
    expect(header.find('.navbar-text .small').text()).toEqual(
      'Trade history viewer web app',
    );
  });
});
