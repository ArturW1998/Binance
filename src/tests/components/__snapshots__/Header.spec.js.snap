// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders properly 1`] = `
"<header>
  <nav className=\\"navbar navbar-expand-sm navbar-dark bg-dark\\">
    <Link className=\\"navbar-brand\\" to=\\"/\\">
      <img src=\\"logo.svg\\" height=\\"25\\" alt=\\"Binance Logo\\" />
    </Link>
    <div className=\\"navbar-text\\">
      <span className=\\"small text-white\\">
        Trade history viewer web app
      </span>
    </div>
  </nav>
</header>"
`;
